### PR TITLE
Add test for envoyfilter

### PR DESCRIPTION
**Pull request by howardjohn**
Hoping this will detect https://github.com/istio/istio/issues/34356

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Comment by brian-avery** [comment]
/test release-notes_istio
--


<br class="Apple-interchange-newline">

File: tests/integration/pilot/common/routing.go [hunk 1]
```diff
@@ -1,4 +1,6 @@
+//go:build integ
 // +build integ
```
Comment (adiprerepa) [inline tests/integration/pilot/common/routing.go:2]: why this change?

Comment (howardjohn) [reply tests/integration/pilot/common/routing.go:2]: "smart"  ide auto inserting it 🙂will revert

**Review by ramaraochavali — APPROVED** [review]
